docs/painless/painless-guide/painless-execute-script.asciidoc

```diff
@@ -164,3 +164,329 @@ Response:
   "result": 0.8
 }
 --------------------------------------------------
+
+==== Runtime field contexts
+
+Runtime field contexts execute a script used for a runtime field and return results in doc-values order.
+
+[[painless-execute-api-contexts-runtime-fields]]
+.Contexts
+[options="header"]
+|======
+| Name            | Emit Type      | Return Type
+| boolean_field   | boolean        | sorted list of booleans
+| date_field      | long (ms)      | sorted list of longs (ms)
+| double_field    | double         | sorted list of doubles
+| geo_point_field | long (encoded) | sorted list of longs (encoded)
+| ip_field        | String         | sorted list of BytesRef[] (encoded)
+| keyword_field   | String         | sorted list of Strings
+| long_field      | long           | sorted list of longs
+|======
+
+[[painless-execute-api-parameters-runtime-fields]]
+.Parameters
+[options="header"]
+|======
+| Name              | Required  | Default | Description
+| `script`          | yes       | -       | The script to execute.
+| `context`         | yes       | -       | The runtime field context the script should be executed in.
+| `context_setup`   | yes       | -       | Additional parameters to the context.
+|======
+
+Specification of the following parameters is required in `context_setup` for a runtime field context:
+
+document:: Contains the document that will be temporarily indexed in-memory and is accessible from the script.
+index:: The name of an index containing a mapping that is compatible with the document being indexed.
+
+*Examples*
+
+* boolean_field
++
+[source,console]
+----------------------------------------------------------------
+PUT /my-index-boolean
+{
+  "mappings": {
+    "properties": {
+      "b": {
+        "type": "boolean"
+      }
+    }
+  }
+}
+
+POST /_scripts/painless/_execute
+{
+  "script": {
+    "source": "emit(params['p']); emit(doc['b'].value);",
+    "params": {
+      "p": true
+    }
+  },
+  "context": "boolean_field",
+  "context_setup": {
+    "index": "my-index-boolean",
+    "document": {
+      "b": false
+    }
+  }
+}
```

Comment:
would you mind using a more meaningful example?

Reply:
I went with the simplest way to show an emit in a runtime fields script, but I'm definitely happy to provide more meaningful examples. Any suggestions?

Reply:
@jdconrad, we could reuse existing examples for runtime fields. The example from [Mapping a runtime field](https://www.elastic.co/guide/en/elasticsearch/reference/current/runtime-mapping-fields.html) is both useful and meaningful, as are the examples that I'm working on for [field extraction](https://elasticsearch_71596.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/common-script-uses.html#scripting-field-extraction).

Reply:
@lockewritesdocs This is a great suggestion!